### PR TITLE
Makes thin wrapper around Javascript loglevel module.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Creates an interface directly analogous to our Python logging interface for frontend Javascript logging.

## Screenshots (if appropriate)

image

Main downside of this is that the stacktrace for the logging is lost, it will now point to the exact line in `logging.js` where the logging call is made, rather than the line in the JS where the logging call originates.

### CODE REVIEW

File: frontend_build/src/webpack.config.base.js [hunk 1]
```diff
@@ -122,6 +123,9 @@ var config = {
   },
   postcss: function () {
     return [autoprefixer];
+  },
+  node: {
+    __filename: true
```
Comment (rtibbles) [inline frontend_build/src/webpack.config.base.js:129]: Enables the `__filename` properly in webpack builds.

File: frontend_build/src/webpack.config.base.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ var config = {
       {
         test: /\.json$/,
         loader: 'json',
+        exclude: /node_modules/
```
Comment (rtibbles) [inline frontend_build/src/webpack.config.base.js:55]: Don't do this in node_modules, they can do it themselves.

File: kolibri/core/assets/src/asset_loader.js [hunk 3]
```diff
@@ -9,7 +9,7 @@
 const scriptjs = require('scriptjs');
 const loadcss = require('fg-loadcss').loadCSS;
 const onloadcss = require('fg-loadcss/src/onloadCSS');
-const logging = require('loglevel');
+const logging = require('./logging').getLogger(__filename);
```
Comment (rtibbles) [inline kolibri/core/assets/src/asset_loader.js]: Can get the logger in the same way.

Outside of the core app, you should still be able to `require('loglevel')` and get the new logging interface.

File: kolibri/core/assets/src/logging.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+/**
+ * Provides thin wrapper around loglevel to allow a Python like interface for logging.
+ * @module logging
+ */
+
+const logging = require('loglevel');
+
+class Logger {
```
Comment (rtibbles) [inline kolibri/core/assets/src/logging.js:9]: Individual logger objects namespaced.

File: kolibri/core/assets/src/logging.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  enableAll(persist) {
+    this.logger.enableAll(persist);
+  }
+
+  disableAll(persist) {
+    this.logger.disableAll(persist);
+  }
+}
+
+class Logging {
```
Comment (rtibbles) [inline kolibri/core/assets/src/logging.js:47]: Overall interface to the logging module.

**Comment by indirectlylit** [comment]
Yeah - losing the source-mapped line number and error stack trace would be unfortunate. `loglevel` seems to work okay - what's the goal with the wrapper?

image

**Comment by rtibbles** [comment]
The goal with the wrapper is to give equivalence with our Python logging, so that we get the type of log prefixed, timestamp, and source file.

I think we can preserve the loglevel behaviour by forgoing the timestamp (which is done automatically in Firefox, and is available optionally in latest Chrome), so I am going to try that option.

**Comment by rtibbles** [comment]
OK, have modified it to preserve the logline information, at the cost of dynamic info (i.e. timestamping).

Have also added an alias so that people can just do:

`const logging = require('logging').getLogger(__filename);`

And get a logging object that is now namespaced to the local file.

**Comment by rtibbles** [comment]
Added snippet to docs.

**Comment by indirectlylit** [comment]
nice

**Comment by indirectlylit** [comment]
This looks good, and it's great that the line number is preserved.

It would be cool to eventually have a setting which disables the prefix in dev mode, because the info is redundant with what chrome provides and takes up a lot of space on a smaller console:

image

**Pull request merged**